CommonIncludes/ve_avx512.hpp

```diff
@@ -0,0 +1,1941 @@
+			tag_type(typename tag_type::value_base_t(a.value + 10)),
+			tag_type(typename tag_type::value_base_t(a.value + 11)),
+			tag_type(typename tag_type::value_base_t(a.value + 12)),
+			tag_type(typename tag_type::value_base_t(a.value + 13)),
+			tag_type(typename tag_type::value_base_t(a.value + 14)),
+			tag_type(typename tag_type::value_base_t(a.value + 15))) == b;
+	}
+	template<typename tag_type>
+	RELEASE_INLINE mask_vector operator!=(tagged_vector<typename ve_identity<tag_type>::type> b, contiguous_tags_base<tag_type> a) {
+		return tagged_vector<tag_type>(
+			tag_type(typename tag_type::value_base_t(a.value)),
+			tag_type(typename tag_type::value_base_t(a.value + 1)),
+			tag_type(typename tag_type::value_base_t(a.value + 2)),
+			tag_type(typename tag_type::value_base_t(a.value + 3)),
+			tag_type(typename tag_type::value_base_t(a.value + 4)),
+			tag_type(typename tag_type::value_base_t(a.value + 5)),
+			tag_type(typename tag_type::value_base_t(a.value + 6)),
+			tag_type(typename tag_type::value_base_t(a.value + 7)),
+			tag_type(typename tag_type::value_base_t(a.value + 8)),
+			tag_type(typename tag_type::value_base_t(a.value + 9)),
+			tag_type(typename tag_type::value_base_t(a.value + 10)),
+			tag_type(typename tag_type::value_base_t(a.value + 11)),
+			tag_type(typename tag_type::value_base_t(a.value + 12)),
+			tag_type(typename tag_type::value_base_t(a.value + 13)),
+			tag_type(typename tag_type::value_base_t(a.value + 14)),
+			tag_type(typename tag_type::value_base_t(a.value + 15))) != b;
+	}
+
+
+	RELEASE_INLINE fp_vector select(mask_vector mask, fp_vector a, fp_vector b) {
+		return _mm512_mask_blend_ps(mask, b, a);
+	}
+
+	RELEASE_INLINE int_vector select(mask_vector mask, int_vector a, int_vector b) {
+		return _mm512_mask_blend_epi32(mask, b.value, a.value);
+	}
+
+	template<typename T>
+	RELEASE_INLINE tagged_vector<T> select(mask_vector mask, tagged_vector<T> a, tagged_vector<typename ve_identity<T>::type> b) {
+		return tagged_vector<T>(_mm512_mask_blend_epi32(mask, b, a), std::true_type{});
+	}
+
+	RELEASE_INLINE mask_vector is_non_zero(int_vector i) {
+		return i != int_vector();
+	}
+	RELEASE_INLINE mask_vector is_zero(int_vector i) {
+		return i == int_vector();
+	}
+
+	template<typename T>
+	RELEASE_INLINE mask_vector is_valid_index(tagged_vector<T> i) {
+		return i != tagged_vector<T>();
+	}
+	template<typename T>
+	RELEASE_INLINE mask_vector is_invalid(tagged_vector<T> i) {
+		return i == tagged_vector<T>();
+	}
+
+
+	template<typename T>
+	RELEASE_INLINE vbitfield_type load(contiguous_tags<T> e, dcon::bitfield_type const* source) {
+		vbitfield_type output;
+		memcpy(&output.v, source + (e.value / uint32_t(8)), sizeof(uint16_t));
+		return output;
+	}
+	template<typename T>
+	RELEASE_INLINE vbitfield_type load(unaligned_contiguous_tags<T> e, dcon::bitfield_type const* source) {
+		auto overflow = (e.value & 0x7) != 0 ? (source[2 + e.value / uint32_t(8)].v) : 0;
```

Comment:
does this pass the tests? This looks like it will be checking if the value is a multiple of 8, but you want to be checking for a multiple of 16. Similarly, source holds 8 bits per index, but in theory you could need up to 15 bits from the overflow.